sonaranalyzer-dotnet/src/SonarAnalyzer.Common/Rules/MethodOverloadsShouldBeGroupedBase.cs

```diff
@@ -89,11 +90,70 @@ protected List<SyntaxToken>[] GetMisplacedOverloads(SyntaxNodeAnalysisContext c,
                 }
             }
             return misplacedOverloads.Values.Where(x => x.Count > 1).ToArray();
+
+            bool ProcessByInterface(TMemberDeclarationSyntax member, List<MemberInfo> others)
+            {
+                if(groupedByInterface.TryGetValue(member, out var interfaces))
+                {
+                    return interfaces.Length==1 && others.Any(other => FindInterfaces(c.SemanticModel, other.Member).Contains(interfaces.Single()));
+                }
+                return true; // Not member of an interface => process
+            }
+        }
+
+        private Dictionary<TMemberDeclarationSyntax, ImmutableArray<INamedTypeSymbol>> GroupedByInterface(SyntaxNodeAnalysisContext c, IEnumerable<TMemberDeclarationSyntax> members)
+        {
+            var ret = new Dictionary<TMemberDeclarationSyntax, ImmutableArray<INamedTypeSymbol>>();
+            ImmutableArray<INamedTypeSymbol> currentInterfaces, previousInterfaces = ImmutableArray<INamedTypeSymbol>.Empty;
+            TMemberDeclarationSyntax previous = null;
+            foreach (var member in members)
+            {
+                currentInterfaces = FindInterfaces(c.SemanticModel, member);
+                if (currentInterfaces.Intersect(previousInterfaces).Any())
+                {
+                    ret.Add(member, currentInterfaces);
+                    if (previous != null && !ret.ContainsKey(previous))
+                    {
+                        ret.Add(previous, previousInterfaces);
+                    }
+                }
+                previousInterfaces = currentInterfaces;
+                previous = member;
+            }
+            return ret;
+        }
+
+        private ImmutableArray<INamedTypeSymbol> FindInterfaces(SemanticModel semanticModel, TMemberDeclarationSyntax member)
```

Comment:
It would be great to add some comments since it will take a while if in 3 months we will have to update this.